src/main/scala/io/iohk/ethereum/merklepatriciatrie/Node.scala

```diff
@@ -0,0 +1,102 @@
+package io.iohk.ethereum.merklepatriciatrie
+
+import io.iohk.ethereum.rlp.{decode => decodeRLP, encode => encodeRLP, _}
+import io.iohk.ethereum.merklepatriciatrie.MerklePatriciaTrie.HashFn
+
+/**
+  * Trie elements
+  */
+private[merklepatriciatrie] sealed trait Node {
+
+  import MerklePatriciaTrie._
+
+  private var hashedCache: Option[Array[Byte]] = None
```

Comment:
@adamsmo i've just pushed to a test branch how a new approach about this might work. Here https://github.com/input-output-hk/etc-client/compare/feature/patriciaTrie...feature/patriciaTrieAdamChangeSuggestion you can find the branch comparison....please let me know your thoughts about this.